packages/babel-plugin-transform-for-of/test/fixtures/opt/typeannotation/input.js

```diff
@@ -1,4 +0,0 @@
-for (const y of (b: Array<any>)) {}
-function a(b: Array<any>) {
-  for (const y of b) {}
-}
```

Comment:
This test is split to two tests because the former will be optimized but not the latter, which is very confusing.